engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/NadelServiceTypeFilterTransform.kt

```diff
@@ -0,0 +1,144 @@
+package graphql.nadel.enginekt.transform
+
+import graphql.introspection.Introspection
+import graphql.nadel.Service
+import graphql.nadel.ServiceExecutionResult
+import graphql.nadel.enginekt.NadelExecutionContext
+import graphql.nadel.enginekt.blueprint.IntrospectionService
+import graphql.nadel.enginekt.blueprint.NadelOverallExecutionBlueprint
+import graphql.nadel.enginekt.transform.NadelServiceTypeFilterTransform.State
+import graphql.nadel.enginekt.transform.artificial.NadelAliasHelper
+import graphql.nadel.enginekt.transform.query.NadelQueryPath
+import graphql.nadel.enginekt.transform.query.NadelQueryTransformer
+import graphql.nadel.enginekt.transform.result.NadelResultInstruction
+import graphql.nadel.enginekt.transform.result.json.JsonNodeExtractor
+import graphql.nadel.enginekt.util.AnyMap
+import graphql.nadel.enginekt.util.JsonMap
+import graphql.nadel.enginekt.util.queryPath
+import graphql.nadel.enginekt.util.resolveObjectTypes
+import graphql.nadel.enginekt.util.toBuilder
+import graphql.normalized.ExecutableNormalizedField
+import graphql.normalized.ExecutableNormalizedField.newNormalizedField
+
+class NadelServiceTypeFilterTransform : NadelTransform<State> {
+    data class State(
+        val aliasHelper: NadelAliasHelper,
+        val typeNamesOwnedByService: Set<String>,
+        val fieldObjectTypesOwnedByService: List<String>,
+        val overallField: ExecutableNormalizedField,
```

Comment:
is `overallField` a different object than `field` passed in as an argument to `transformField`? Is this a mechanism to handle cases where other transforms can change the original field before this transform is executed?

Reply:
Yup exactly.